src/devices/imuBosch_BNO055/imuBosch_BNO055.cpp

```diff
@@ -436,9 +435,9 @@ void BoschIMU::run()
 
     // TODO: how to optimally protect only code filling the data vector?
 
-    mutex.wait();
-
-    data.zero();
+    // In order to avoid zeros when a single read from a sensor is missing,
+    // initialize the new measure to be equal to the previous one
+    yarp::sig::Vector data_tmp = data;
```

Comment:
Personal Nitpick : the creation of a `yarp::sig::Vector` involves dynamical allocation of memory, that is tipically a costly a non-realtime safe operation. For vector of this sizes the dynamic the performance hit is probably negligible also due to the optimization in the memory allocator. Anyhow for cleanness I typically allocate buffers only once (i.e. as class members), but feel free to due what you seem more fit.